src/components/SelectionScreen.tsx

```diff
@@ -95,24 +118,33 @@ function SelectionScreen({
             shouldBeBlocked={isConnectionEmpty || shouldBeBlocked}
         >
             <ScreenWrapper
-                includeSafeAreaPaddingBottom={false}
+                includeSafeAreaPaddingBottom={!!errors && !isEmptyObject(errors)}
                 testID={displayName}
             >
                 <HeaderWithBackButton
                     title={translate(title)}
                     onBackButtonPress={onBackButtonPress}
                 />
-                <SelectionList
-                    onSelectRow={onSelectRow}
-                    headerContent={headerContent}
-                    sections={sections}
-                    ListItem={listItem}
-                    showScrollIndicator
-                    shouldShowTooltips={false}
-                    initiallyFocusedOptionKey={initiallyFocusedOptionKey}
-                    listEmptyContent={listEmptyContent}
-                    listFooterContent={listFooterContent}
-                />
+                {headerContent}
+                <OfflineWithFeedback
+                    pendingAction={pendingAction}
+                    errors={errors}
+                    errorRowStyles={[errorRowStyles]}
+                    onClose={onClose}
+                    style={[styles.flex1]}
+                    contentContainerStyle={[styles.flex1]}
+                >
```

Comment:
Interesting, should this be closer to the list options above? How does this error get triggered in the first place?

Reply:
In this case, the error will show when user selected a card but the api was failing.

https://github.com/Expensify/App/assets/16502320/4e7218f8-62b7-4262-bef2-f88d1ee394e8

Reply:
Wow, I think we need to clean up the spacing here. For example, I would expect there to be at least 12px of space below this error message and above the row below it:
![CleanShot 2024-07-10 at 09 13 16@2x](https://github.com/Expensify/App/assets/2319350/32f1de40-032f-422c-89bc-bbf960bb3378)

Reply:
cc @war-in

Reply:
The error at the bottom feels strange, it doesn't feel connected to anything. Can we make it closer to the list options? Or even above them?

Reply:
We already have RBR if we have error during selecting card, should we still need the bottom error?

Reply:
@hungvu193 The error could have some important information about what went wrong (however it doesn't now) so I think it can be useful to display it

@shawnborton We use `OfflineWithFeedback` to handle errors in the app and it puts errors under the connected component. I've tried to put it just under the list but it's a bit tricky (a lot of playing with css). If you think the best option is to put error just below the list I can give it another try (hopefully will make it work this time...)

Reply:
Sounds good, if it's too tricky we can just keep it where it's currently at.

Reply:
But TBH in other integrations (e.g. NetSuite) we don't display errors on SelectionScreen. So maybe we should stick to that approach 🤔

Reply:
Otherwise we probably should update other integrations to display those errors too

Reply:
Good news here. I found the way of doing that (sticking error to the bottom of the list) but we would have to omit `OfflineWithFeedback` usage.

My vision:
- create a new optional prop in `SelectionList` for passing `flexGrow: 0` to the `style` property of `SectionList`
- take out the logic of making an error from `OfflineWithFeedback` (just instantiating the `MessagesRow` component)
- pass `MessagesRow` component as child to `SelectionList`

And that should be it. I think I'll prepare PR in our fork to demonstrate the code and behaviour. Then we can decide if we want to use it

Reply:
Cool, I don't feel too strongly since this is a bit of an edge case anyways. I think we should just stay consistent with what we're doing elsewhere for now.

Reply:
Please take a look on [this PR](https://github.com/software-mansion-labs/expensify-app-fork/pull/93) 🙏 
This way we could `easily` add error handling to other integrations

Reply:
Definitely looks like a big improvement to me! cc @Expensify/design @trjExpensify @mountiny for thoughts there

Reply:
What is the drawback of not using OfflineWithFeedback? Wouldnt that mean we will not get the correct pattern B behaviour?

Reply:
We actually can use `OfflineWithFeedback` but only for showing the offline state. To display errors we need to copy that behaviour into `SelectionScreen` and that's the drawback. We will have some duplicated code (we can of course extract it to separate component)

Reply:
It does seem much better with the error closer to the list.

Reply:
> that's the drawback. We will have some duplicated code (we can of course extract it to separate component)

Yeah I think it would be great to further explore how to dry this up, from design perspective I agree it looks better

Reply:
@war-in Can you fix this [spacing](https://github.com/Expensify/App/pull/44739#discussion_r1671739301) in other places as well? (ie: Advance page).

Reply:
@mountiny Do you think we need separate issue for adjusting error handling in other integrations? I can take care of it once this PR is merged

Reply:
@mountiny Kind bump :)